build.sbt

```diff
@@ -112,14 +178,28 @@ lazy val mdoc = project
       scalaVersion,
       scalaBinaryVersion
     ),
+    libraryDependencies ++= crossSetting(
+      scalaVersion.value,
+      if3 = List(
+        "ch.epfl.lamp" %% "dotty-compiler" % scalaVersion.value,
+        ("org.scalameta" %% "scalameta" % V.scalameta)
+          .withDottyCompat(scalaVersion.value),
+        ("com.geirsson" %% "metaconfig-typesafe-config" % "0.9.10")
```

Comment:
Most usages of metaconfig relate to CLI, which is not supported currently, so most of the related code is removed from Scala 3 jar. We also need to migrate it to Scala 3 if we want to be able to cross compile everything by default.